docs/deployment/resource-providers/mesos.md

```diff
@@ -201,51 +208,50 @@ try (FileOutputStream output = new FileOutputStream(jobGraphFile);
+   in that directory. The user code jar will be loaded by the user code classloader after launching 
+   a Per-Job Cluster via `${FLINK_HOME}/bin/mesos-appmaster-job.sh ...`.
+
+### Session Mode
 
-1. One way is putting them in the `lib/` directory, which will result in the user code jar being loaded by the system classloader.
-1. The other way is creating a `usrlib/` directory in the parent directory of `lib/` and putting the user code jar in the `usrlib/` directory.
-After launching a job cluster via `bin/mesos-appmaster-job.sh ...`, the user code jar will be loaded by the user code classloader.
+The [Getting Started](#starting-a-flink-session-on-mesos) guide at the top of this page describes 
+deploying Flink in Session Mode.
 
-#### General configuration
+## Flink on Mesos Reference
 
-It is possible to completely parameterize a Mesos application through Java properties passed to the Mesos application master.
-This also allows to specify general Flink configuration parameters.
-For example:
+### Configuring Flink on Mesos
 
-    bin/mesos-appmaster.sh \
-        -Dmesos.master=master.foobar.org:5050 \
-        -Djobmanager.memory.process.size=1472m \
-        -Djobmanager.rpc.port=6123 \
-        -Drest.port=8081 \
-        -Dtaskmanager.memory.process.size=3500m \
-        -Dtaskmanager.numberOfTaskSlots=2 \
-        -Dparallelism.default=10
+For a list of Mesos-specific configuration parameters, refer to the 
+[Mesos section]({% link deployment/config.md %}#mesos) of the configuration documentation.
 
-### High Availability
+### High Availability on Mesos
 
-You will need to run a service like Marathon or Apache Aurora which takes care of restarting the JobManager process in case of node or process failures.
-In addition, Zookeeper needs to be configured like described in the [High Availability section of the Flink docs]({% link deployment/ha/index.md %}).
+You will need to run a service like Marathon or Apache Aurora which takes care of restarting the 
+JobManager process in case of node or process failures. In addition, Zookeeper needs to be 
+configured as described in the 
+[High Availability section of the Flink docs]({% link deployment/ha/index.md %}).
 
 #### Marathon
 
-Marathon needs to be set up to launch the `bin/mesos-appmaster.sh` script.
-In particular, it should also adjust any configuration parameters for the Flink cluster.
+Marathon needs to be set up to launch the `bin/mesos-appmaster.sh` script. In particular, it should 
+also adjust any configuration parameters for the Flink cluster.
 
 Here is an example configuration for Marathon:
 
+{% highlight javascript %}
     {
         "id": "flink",
         "cmd": "$FLINK_HOME/bin/mesos-appmaster.sh -Djobmanager.memory.process.size=1472m -Djobmanager.rpc.port=6123 -Drest.port=8081 -Dtaskmanager.memory.process.size=1024m -Dtaskmanager.numberOfTaskSlots=2 -Dparallelism.default=2 -Dmesos.resourcemanager.tasks.cpus=1",
         "cpus": 1.0,
         "mem": 1024
     }
```

Comment:
I think this yaml is not correct. We allocate `mem: 1024` but set `-Djobmanager.memory.process.size=1472m`. Maybe we can put the simplest definition possible here (e.g. w/o specifying the resources and the ports, for example) or we completely remove it and refer to the documentation of Marathon.